src/components/BaseLayer.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useEffect, useState } from 'react'
+import { TileLayer } from '@datapunt/react-maps'
+import { TileLayer as TileLayerType } from 'leaflet'
+import { DEFAULT_AMSTERDAM_LAYERS } from '../constants'
+
+type Props = {
```

Comment:
An `interface` is generally preferred over a `type` for this, as it has more capabilities. See: https://stackoverflow.com/questions/37233735/typescript-interfaces-vs-types